### PR TITLE
OpenTitan: Fix broken UART and missing test cases

### PR DESCRIPTION
**Pull request by alistair23**
### Pull Request Overview

A few recent commits have broken OpenTitan on the CW310 FPGA. This PR fixes the broken UART config and adds tests that were previously removed.

### Testing Strategy

Running unit tests on the FPGA

### Documentation Updated

- [X] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [X] Ran `make prepush`.

### CODE REVIEW

**Review by twilfredo — APPROVED** [review]

File: boards/opentitan/src/tests/flash.rs [hunk 1]
```diff
@@ -2,417 +2,474 @@
...
-    fn write_complete(&self, page: &'static mut F::Page, result: Result<(), hil::flash::Error>) {
-        if self.write_pending.get() {
-            if result.is_err() {
-                let error = result.unwrap_err();
-                assert_eq!(error, hil::flash::Error::FlashMemoryProtectionError);
-                self.mp_fault_detect.set(true);
+    impl<'a, F: hil::flash::Flash> hil::flash::Client<F> for FlashCtlCallBack {
+        fn read_complete(&self, page: &'static mut F::Page, error: Result<(), hil::flash::Error>) {
+            if self.read_pending.get() {
+                if error == Err(hil::flash::Error::FlashMemoryProtectionError) {
```
Comment (nbdd0121) [inline boards/opentitan/src/tests/flash.rs:121]:
```suggestion
                if let Err(hil::flash::Error::FlashMemoryProtectionError) = err {
```

Comment (alistair23) [reply boards/opentitan/src/tests/flash.rs:121]: Is this an improvement? We aren't using the error variable so I don't see this being any cleaner

**Review by bradjc — APPROVED** [review]

**Comment by alistair23** [comment]
Ping!

**Comment by lschuermann** [comment]
Sorry for the delay on this one! I was using this as an excuse to test my CW310 dev infrastructure, but that took a little longer to set up.

I can confirm that without these changes, the UART doesn't work (no `OpenTitan initialisation complete. Entering main loop`), and with these changes it does (at least on [b610a0e12613](https://github.com/lowrisc/opentitan/tree/b610a0e126130b4d81d5961f81104313292542e1)). However, does this fix also work on the pinned revision of OpenTitan we target here?

I'm having trouble recalling some of these conversations, but I thought that Tock was -- for now -- targeting the stable OT release. Maybe I'm mistaken, but in this case this PR should presumably also bump the OT revision, right?

**Comment by lschuermann** [comment]
Oh, are you referring to commits _in Tock_ that broke UART on OpenTitan? That would explain the behavior I'm seeing. If you could clarify that, I'd be happy to merge. :smiley:

**Comment by alistair23** [comment]
> Oh, are you referring to commits _in Tock_ that broke UART on OpenTitan? That would explain the behavior I'm seeing. If you could clarify that, I'd be happy to merge. 😃

Yes, commits in Tock

**Comment by lschuermann** [comment]
@jrvanwhy mentioned that there are some more issues with the Pinmux configuration beyond this particular fix. I guess that this is a net improvement still, so we can probably merge. We should document the issues mentioned by @jrvanwhy (or simply fix them) though.

**Review by lschuermann — APPROVED** [review]

**Comment by alistair23** [comment]
Ping!

**Pull request merged**